crates/runtime/src/utils.rs

```diff
@@ -0,0 +1,26 @@
+/*
+Copyright 2024-2025 The Spice.ai OSS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+pub(crate) fn is_address_in_use_error(err: &tonic::transport::Error) -> bool {
```

Comment:
I dislike having a `utils` module - it usually means you haven't thought enough about where the function should live. This is very specific to the tonic servers, so I would actually just duplicate this into both `crates/runtime/src/flight/mod.rs` and `crates/runtime/src/opentelemetry.rs` for now.

We should probably split the servers into their own crate (i.e. `runtime-servers`) that this could live in - but for now let's just duplicate the code - its very small and unlikely to change.